### PR TITLE
[BEAM-782] support runner names to be partial or case insensitive

### PR DESCRIPTION
**Pull request by sb2nov**
R: @aaltay PTAL

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

File: sdks/python/apache_beam/runners/runner.py [hunk 1]
```diff
@@ -26,9 +26,22 @@
 import tempfile
 
 
+_path_generator = lambda rs, p: {r.lower(): p + r for r in rs}
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/runner.py]: This should be a proper function, with a clear name and clearly named arguments.

File: sdks/python/apache_beam/runners/runner.py [hunk 2]
```diff
@@ -47,12 +60,11 @@ def create_runner(runner_name):
     RuntimeError: if an invalid runner name is used.
   """
 
-  if runner_name in _KNOWN_DIRECT_RUNNERS:
-    runner_name = 'apache_beam.runners.direct.direct_runner.' + runner_name
-  elif runner_name in _KNOWN_DATAFLOW_RUNNERS:
-    runner_name = 'apache_beam.runners.dataflow_runner.' + runner_name
-  elif runner_name in _KNOWN_TEST_RUNNERS:
-    runner_name = 'apache_beam.runners.test.' + runner_name
+  # Get the qualified runner name by matching the lower case of the name first
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/runner.py]: How about

```
# Get the qualified runner name by using the lower case runner name. If that
# fails try appending the name with 'runner' and check if it matches. If that also fails,
# use the given runner name as is.
```

File: sdks/python/apache_beam/runners/runner.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/runners/runner.py]: Done.

File: sdks/python/apache_beam/runners/runner.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/runners/runner.py]: Done.

**Comment by aaltay** [comment]
LGTM. Thank you.

**Pull request closed without merging**